services/server/server.go

```diff
@@ -515,6 +540,29 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 			DisableSTT:             req.DisableStt,
 			PMMPublicAddress:       req.PmmPublicAddress,
 			RemovePMMPublicAddress: req.RemovePmmPublicAddress,
+
+			EnableAlerting:              req.EnableAlerting,
+			DisableAlerting:             req.DisableAlerting,
+			RemoveEmailAlertingSettings: req.RemoveEmailAlertingSettings,
+			RemoveSlackAlertingSettings: req.RemoveSlackAlertingSettings,
+		}
+
+		if req.EmailAlertingSettings != nil {
+			settingsParams.EmailAlertingSettings = &models.EmailAlertingSettings{
+				From:      req.EmailAlertingSettings.GetFrom(),
+				Smarthost: req.EmailAlertingSettings.GetSmarthost(),
+				Hello:     req.EmailAlertingSettings.GetHello(),
+				Username:  req.EmailAlertingSettings.GetUsername(),
+				Password:  req.EmailAlertingSettings.GetPassword(),
```

Comment:
Since we never return password in the API (due to our security model I'm working on now), we should not blindly override password there. Instead, we should set only if it not empty.

(right now, it will not possible be possible to remove a password)

Reply:
And please add API test for that

Reply:
do you mean API test for checking that an empty password was not set? how will I check that? we always return an empty string for password from the API

Reply:
That code there would overwrite the password, right? That was a bug that is fixed now. So we should add a test for it